src/Connection/Socket/Socket.php

```diff
@@ -0,0 +1,270 @@
+<?php
```

Comment:
Missing PHP7 return types/type hints

Reply:
Unfortunately the PSR-7 StreamInterface has no strict typing.

Reply:
That doesn't stop you from using strict return types ;)